packages/contracts-bedrock/contracts/test/L1ERC721Bridge.t.sol

```diff
@@ -0,0 +1,402 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { Messenger_Initializer } from "./CommonTest.t.sol";
+import { L1ERC721Bridge } from "../L1/L1ERC721Bridge.sol";
+import { L2ERC721Bridge } from "../L2/L2ERC721Bridge.sol";
+
+contract TestERC721 is ERC721 {
+    constructor() ERC721("Test", "TST") {}
+
+    function mint(address to, uint256 tokenId) public {
+        _mint(to, tokenId);
+    }
+}
+
+contract L1ERC721Bridge_Test is Messenger_Initializer {
+    TestERC721 internal localToken;
+    TestERC721 internal remoteToken;
+    L1ERC721Bridge internal bridge;
+    address internal constant otherBridge = address(0x3456);
+    uint256 internal constant tokenId = 1;
+
+    event ERC721BridgeInitiated(
+        address indexed localToken,
+        address indexed remoteToken,
+        address indexed from,
+        address to,
+        uint256 tokenId,
+        bytes extraData
+    );
+
+    event ERC721BridgeFinalized(
+        address indexed localToken,
+        address indexed remoteToken,
+        address indexed from,
+        address to,
+        uint256 tokenId,
+        bytes extraData
+    );
+
+    function setUp() public override {
+        super.setUp();
+
+        // Create necessary contracts.
+        bridge = new L1ERC721Bridge(address(L1Messenger), otherBridge);
```

Comment:
Can you label the bridge so we get nice traces?

Reply:
Done